internal/process/provisioning/create_runtime_resource_step.go

```diff
@@ -298,13 +298,21 @@ func (s *CreateRuntimeResourceStep) createKubernetesConfiguration(operation inte
 		}
 	}
 
-	return imv1.Kubernetes{
-		Version: ptr.String(s.config.KubernetesVersion),
-		KubeAPIServer: imv1.APIServer{
-			OidcConfig:           oidc,
-			AdditionalOidcConfig: nil,
-		},
+	kubernetesConfig := imv1.Kubernetes{
+		Version:       ptr.String(s.config.KubernetesVersion),
+		KubeAPIServer: imv1.APIServer{},
+	}
+
+	if s.config.UseMainOIDC {
```

Comment:
So we have two separate settings, hence four combinations are possible?

Reply:
Yes, but the scenario where both flags are false shouldn't happen. It is in our responsibility to not set both to false. I can add a check for this if you think we need this.